components/pagination/skeleton_view.coffee

```diff
@@ -1,148 +1,148 @@
+    @resetQueue toLoad
 
-  addPage: (id)->
-    @pages[id] =
-      loaded: false
-      loading: false
-      xhr: {}
-      num: id
-
-  get: (id) -> @pages[id]
+  resetQueue: (items = []) ->
+    @queue = items
 
-  pageLoading: (id) ->
-    @pages[id].loading = true
+    @cancelStaleRequests()
 
-  pageLoaded: (id) ->
-    @pages[id].loaded = true
-    --@loading
-    @startLoad()
+    @loading = _.filter(@pages, ({ loading }) => loading).length
 
-  resetQueue: (items=[]) ->
-    @queue = items
-    @cancelStaleRequests()
-    @loading = _.filter(@pages, (page)=>page.loading).length
     @startLoad()
 
   cancelStaleRequests: ->
     for page, key in @pages
-      if page.loading and not _.include(@queue, page.num)
-        @cancelRequest(page.num)
-
-  cancelAllRequests: ->
-    _.each @pages, (page) =>
-      @cancelRequest(page.num)
+      if page.loading and not _.include(@queue, page.number)
+        @cancelRequest page.number
 
   startLoad: ->
     (load = =>
-      return unless @queue.length and (@loading < 2)
+      return unless @queue.length and @loading < 2
 
       current = @queue.shift()
 
-      if @pages[current] and not @pages[current].loading and not @pages[current].loaded
+      if @pages[current] and (not @pages[current].loading) and (not @pages[current].loaded)
         ++@loading
 
-        @setPlaceholdersLoading(current)
-
-        @pages[current] =
-          xhr: @collection.loadPage current, @
+        @pages[current] = _.extend @pages[current],
+          xhr: @collection.loadPage current, this
           loaded: false
           loading: true
-          num: current
 
       load()
     )()
 
-  setPlaceholdersLoading: (id) -> $("#page-#{id}").nextUntil("#page-#{id + 1}").addClass('loading')
-
-  unsetPlaceholdersLoading: (id) -> $("#page-#{id}").nextUntil("#page-#{id + 1}").removeClass('loading')
-
-  completeRequest: (page_id) ->
-    @pagesLoaded()
-    @markPageLoaded(page_id)
+  completeRequest: (id) ->
+    @markPageLoaded id
     @startLoad()
 
-    mediator.trigger 'page:loaded'
-    mediator.shared.blocks = @collection
-
-    # $start = if page_id is 1 then @$('> .grid__block:first') else @$(".pagemarker[data-page=#{page_id}]")
-    # $pageElements = $start.nextUntil('.pagemarker').andSelf()
+    if waypoints = @pages[id].waypoints
+      waypoints.top.destroy()
+      waypoints.bottom.destroy()
 
-    # $pageElements = $pageElements.filter('.grid__block--placeholder')
-
-    # _.delay =>
-    #   $pageElements.remove()
-    # , 100
+    mediator.shared.blocks = @collection
 
   markPageLoaded: (id) ->
     @pages[id]?.loaded = true
     @pages[id]?.loading = false
-    --@loading
-
-  pagesLoaded: ->
-    _.each @pages, (page)->
-      return false unless page.loaded
 
-    return true
+    --@loading
 
-  cancelRequest: (id)->
+  cancelRequest: (id) ->
     @pages[id]?.xhr?.abort?()
     @pages[id]?.loaded = false
     @pages[id]?.loading = false
-    @unsetPlaceholdersLoading(id)
```

Comment:
A lot of unneeded complexity here that I got rid of. This implements  the pagination without the need for those weird pagemarkers. This also removes the waypoints once they are used for a less janky scroll.